### PR TITLE
[9.0]FIX #463 : Use absolute path to prevent some deployements issues

### LINKED ISSUE #463: [9.0]pain xsd not found as an absolute path
Hi,

In v9 we have a strange issue while trying to generate the payment file :
```
n call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python2.7/site-packages/openerp/addons/web/controllers/main.py", line 887, in _call_kw
    return getattr(request.registry.get(model), method)(request.cr, request.uid, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 250, in wrapper
    return old_api(self, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 381, in old_api
    result = method(recs, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_payment_order/models/account_payment_order.py", line 312, in open2generated
    payment_file_str, filename = self.generate_payment_file()
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_banking_sepa_direct_debit/models/account_payment_order.py", line 39, in generate_payment_file
    return super(AccountPaymentOrder, self).generate_payment_file()
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_banking_sepa_credit_transfer/models/account_payment_order.py", line 163, in generate_payment_file
    return self.finalize_sepa_file_creation(xml_root, gen_args)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_banking_sepa_direct_debit/models/account_payment_order.py", line 298, in finalize_sepa_file_creation
    xml_root, gen_args)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_banking_pain_base/models/account_payment_order.py", line 157, in finalize_sepa_file_creation
    self._validate_xml(xml_string, gen_args)
  File "/usr/lib/python2.7/site-packages/openerp/api.py", line 248, in wrapper
    return new_api(self, *args, **kwargs)
  File "/opt/odoo/additional_repos/OCA-bank-payment/account_banking_pain_base/models/account_payment_order.py", line 128, in _validate_xml
    tools.file_open(gen_args['pain_xsd_file']))
  File "/usr/lib/python2.7/site-packages/openerp/tools/misc.py", line 188, in file_open
    pathinfo=pathinfo, basename=basename)
  File "/usr/lib/python2.7/site-packages/openerp/tools/misc.py", line 206, in _fileopen
    raise ValueError("Unknown path: %s" % name)
ValueError: Unknown path: /account_banking_sepa_credit_transfer/data/pain.001.001.03.xsd

```
What is strange is that a first file have been generated successfully and ther is no reason to address the xsd as an absolute path from the root of the server instead of the root of the module.

Any advice would be appreciated

### PR DESCRIPTION
**Pull request by flotho**

### CODE REVIEW

**Comment by flotho** [comment]
Hi @pedrobaeza ,

I declare this PR ready for review

Hope it could help

File: account_banking_sepa_credit_transfer/models/account_payment_method.py [hunk 1]
```diff
@@ -23,7 +23,8 @@ def get_xsd_file_path(self):
         if self.pain_version in [
                 'pain.001.001.02', 'pain.001.001.03', 'pain.001.001.04',
                 'pain.001.001.05', 'pain.001.003.03']:
-            path = 'account_banking_sepa_credit_transfer/data/%s.xsd'\
-                % self.pain_version
+            abs_path = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
```
Comment (pedrobaeza) [inline account_banking_sepa_credit_transfer/models/account_payment_method.py:26]: This is not the way. There's a method in odoo.tools for getting the correct path.

Comment (flotho) [reply account_banking_sepa_credit_transfer/models/account_payment_method.py:26]: Hi @pedrobaeza , 

humm, I was expecting this but I didn't found this or maybe did I make a wrong interpretation of methods here :
* https://github.com/odoo/odoo/blob/9.0/openerp/tools/appdirs.py 

Actually or deployment is done via multiple paths in addons_path and i'm not sure to find the correct method for getting the module path itself.
Did I searched in a wrong file?

regards

Comment (pedrobaeza) [reply account_banking_sepa_credit_transfer/models/account_payment_method.py:26]: The method is `tools.file_open` with the relative path and it returns a class that encloses the opened file. Some other methods like `tools.convert_xml_import` directly accept that class.

**Comment by StefanRijnhart** [comment]
Any chance of updating this @flotho ? Looks like a good fix.

**Pull request closed without merging**